#! /bin/sh -e

# DP: fixes to build with gcc-4.0 on amd64

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        #cd ${dir}gcc && autoconf
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        #rm ${dir}gcc/configure
        ;;
    *)
	echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
        exit 1
esac
exit 0

diff -urN ../tmp-orig/isdnutils-3.6.2005-01-03/isdnlog/tools/isdnrate.c ./isdnlog/tools/isdnrate.c
--- ../tmp-orig/isdnutils-3.6.2005-01-03/isdnlog/tools/isdnrate.c	2004-12-18 16:10:50.000000000 +0100
+++ ./isdnlog/tools/isdnrate.c	2005-03-03 22:20:56.443407426 +0100
@@ -327,7 +327,7 @@
 
 static void print_header(void);
 
-static char *myname, *myshortname;
+char *myname,*myshortname;
 static char options[] = "ab:c:d:f:h:i:l:op:r:st:v::x:CD::G:HLNP:O:S:TUVX::Z";
 static char usage[] = "%s: usage: %s [ -%s ] Destination ...\n";
 
@@ -336,7 +336,7 @@
 static int usestat = 0;
 static int duration = LCR_DURATION;
 static time_t start;
-static int day, month, year, hour, min, sec;
+static int year, min, sec;
 static char ignore[MAXPROVIDER];
 static char *fromarea = 0;
 static char wanted_day;
diff -urN ../tmp-orig/isdnutils-3.6.2005-01-03/eicon/eiconctrl.c ./eicon/eiconctrl.c
--- ../tmp-orig/isdnutils-3.6.2005-01-03/eicon/eiconctrl.c	2001-03-01 15:59:12.000000000 +0100
+++ ./eicon/eiconctrl.c	2005-03-03 22:20:10.278318140 +0100
@@ -1970,7 +1970,7 @@
 				val = strtol(p, &q, 16);
 				p = q;
 				val = strtol(p, &q, 16);
-				(unsigned short) *buffer = (unsigned short) val;
+				*buffer = (unsigned short) val;
 				pos = 2;
 				while ((p != q) && (*q != 0)) {
 					p = q;
Binary files ../tmp-orig/isdnutils-3.6.2005-01-03/hisax/ISAR.BIN and ./hisax/ISAR.BIN differ
diff -urN ../tmp-orig/isdnutils-3.6.2005-01-03/hisax/hisaxctrl.c ./hisax/hisaxctrl.c
--- ../tmp-orig/isdnutils-3.6.2005-01-03/hisax/hisaxctrl.c	2000-06-29 19:38:26.000000000 +0200
+++ ./hisax/hisaxctrl.c	2005-03-03 22:18:37.907147481 +0100
@@ -43,7 +43,7 @@
 		fprintf(stderr, "cannot open file %s\n", fname);
 		exit(-1);
 	}
-	p = (int *) buffer = (unsigned char *) malloc(MAX_SIZE+4);
+	p = buffer = (unsigned char *) malloc(MAX_SIZE+4);
 	if (!buffer) {
 		fprintf(stderr, "cannot get %d byte memory\n", MAX_SIZE+4);
 		exit(-1);
diff -urN ../tmp-orig/isdnutils-3.6.2005-01-03/ipppd/main.c ./ipppd/main.c
--- ../tmp-orig/isdnutils-3.6.2005-01-03/ipppd/main.c	2003-07-01 00:30:57.000000000 +0200
+++ ./ipppd/main.c	2005-03-03 22:21:23.684149461 +0100
@@ -1044,10 +1044,10 @@
 static char line[256];			/* line to be logged accumulated here */
 static char *linep;
 
+static void pr_log __P((void *, char *, ...));
+
 void log_packet(u_char *p,int len,char *prefix,int linkunit)
 {
-	static void pr_log __P((void *, char *, ...));
-
     int i, n;
     u_short proto;
     u_char x;


